packages/nextjs/src/server/clerkMiddleware.ts

```diff
@@ -465,7 +465,7 @@ const handleControlFlowErrors = (
   requestState: RequestState,
 ): Response => {
   if (isNextjsUnauthorizedError(e)) {
-    const response = NextResponse.next({ status: 401 });
+    const response = NextResponse.json({ message: 'Unauthorized' }, { status: 401 });
```

Comment:
We need to terminate requests for invalid tokens. The previous one continues processing the middleware chain instead of ending the request.